iter/itergen.go

```diff
@@ -0,0 +1,37 @@
+package iter
+
+import (
+	"github.com/dgryski/go-tsz"
+	"github.com/raintank/worldping-api/pkg/log"
+)
+
+type IterGen struct {
+	b   []byte
+	ts  uint32
+	len uint64
```

Comment:
instead of having a len attribute and expecting callers of NewGen to provide it,  can't we simply have the Length() method return `len(ig.b)` ?